api/provider/aws/test_aws/service_create_test.go

```diff
@@ -0,0 +1,246 @@
+	createServiceInput.SetCluster("l0-test-env_id")
+	createServiceInput.SetDesiredCount(1)
+	createServiceInput.SetServiceName("l0-test-svc_id")
+	createServiceInput.SetTaskDefinition("dpl_arn")
+
+	loadBalancer := &ecs.LoadBalancer{}
+	loadBalancer.SetContainerName("ctn_name")
+	loadBalancer.SetContainerPort(80)
+	loadBalancer.SetLoadBalancerName("lb_name")
+
+	loadBalancers := []*ecs.LoadBalancer{loadBalancer}
+
+	createServiceInput.SetLoadBalancers(loadBalancers)
+	createServiceInput.SetRole("l0-test-lb_id-lb")
+
+	mockAWS.ECS.EXPECT().
+		CreateService(createServiceInput).
+		Return(&ecs.CreateServiceOutput{}, nil)
+
+	req := models.CreateServiceRequest{
+		DeployID:       "dpl_id",
+		EnvironmentID:  "env_id",
+		LoadBalancerID: "lb_id",
+		ServiceName:    "svc_name",
+	}
+
+	target := provider.NewServiceProvider(mockAWS.Client(), tagStore, mockConfig)
+	result, err := target.Create(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, result, "svc_id")
+
+	expectedTags := models.Tags{
+		{
+			EntityID:   "svc_id",
+			EntityType: "service",
+			Key:        "name",
+			Value:      "svc_name",
+		},
+		{
+			EntityID:   "svc_id",
+			EntityType: "service",
+			Key:        "environment_id",
+			Value:      "env_id",
+		},
+		{
+			EntityID:   "svc_id",
+			EntityType: "service",
+			Key:        "load_balancer_id",
+			Value:      "lb_id",
+		},
+	}
+
+	for _, tag := range expectedTags {
+		assert.Contains(t, tagStore.Tags(), tag)
+	}
+}
+
+func TestServiceCreate_defaults(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	defer provider.SetEntityIDGenerator("svc_id")()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "dpl_id",
+			EntityType: "deploy",
+			Key:        "arn",
+			Value:      "dpl_arn",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// ServiceProvider.Create() makes AWS calls that need to be mocked out:
+	// - ELB.DescribeLoadBalancer (if req.LoadBalancerID != "")
+	// - ECS.DescribeTaskDefinition (if req.LoadBalancerID != "")
+	// - ECS.CreateService
+
+	// ECS.CreateService
+	createServiceInput := &ecs.CreateServiceInput{}
+	createServiceInput.SetCluster("l0-test-env_id")
+	createServiceInput.SetDesiredCount(1)
+	createServiceInput.SetServiceName("l0-test-svc_id")
+	createServiceInput.SetTaskDefinition("dpl_arn")
+
+	mockAWS.ECS.EXPECT().
+		CreateService(createServiceInput).
+		Return(&ecs.CreateServiceOutput{}, nil)
+
+	req := models.CreateServiceRequest{
+		DeployID:      "dpl_id",
+		EnvironmentID: "env_id",
+		ServiceName:   "svc_name",
+	}
+
+	target := provider.NewServiceProvider(mockAWS.Client(), tagStore, mockConfig)
+	result, err := target.Create(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, result, "svc_id")
```

Comment:
we don't need to check this behavior since it was already done in the non-default test

Reply:
✅